Application/EdFi.Ods.Common/Infrastructure/Activities/ParameterListSetter.cs

```diff
@@ -23,5 +24,24 @@ public void SetParameterList(IQuery query, string name, IEnumerable ids)
         {
             query.SetParameterList(name, ids);
         }
+
+        /// <summary>
+        /// Apply an "in" constraint to the named property.
+        /// </summary>
+        /// <param name="propertyName">
+        /// The name of the Property in the class.
+        /// </param>
+        /// <param name="values">An array of values.</param>
+        /// <returns>An <see cref="AbstractCriterion" />.</returns>
+        public AbstractCriterion In(string propertyName, object[] values)
+        {
+            // Nhibernate's Restrictions.In() does not expect view aliases to be enclosed in
+            // curly brackets (contrary to Expression.Sql())
+            propertyName = propertyName
```

Comment:
Since this code could be executed quite frequently, it could result in many unnecessary string allocations. Given that there is a fairly limited set of property names for which this would be repeatedly calculated, I think I would recommend save the result in a `ConcurrentDictionary<string, string>` using the `GetOrAdd` method with a factory method.

For the factory method, a few string allocations are probably not a big deal (using the code above), but you could also use a span here to perform 0 or 1 string allocation. The logic below code is definitely less readable, but I was already exploring it when I had the thought about just storing the result in a dictionary so I'm including it here anyway. So long as the string allocations only happen the first time a property name is passed in, I'm good with either implementation.

```csharp
var propertyNameSpan = propertyName.AsSpan();

int openBraceIndex = propertyNameSpan.IndexOf('{');
while (propertyNameSpan[openBraceIndex + 1] == '{') { openBraceIndex++; }

int closeBraceIndex = propertyNameSpan.LastIndexOf('}');
while (closeBraceIndex > 0 && propertyNameSpan[closeBraceIndex - 1] == '}') { closeBraceIndex--; }

if (closeBraceIndex >= 0)
{
    return propertyNameSpan.Slice(openBraceIndex + 1, closeBraceIndex - openBraceIndex - 1).ToString();
}
else if (openBraceIndex >= 0)
{
    return propertyNameSpan.Slice(openBraceIndex + 1).ToString();
}
else
{
    return propertyName;
}
```

Reply:
I added a `ConcurrentDictionary` with a factory method that creates string allocations because it's more readable.